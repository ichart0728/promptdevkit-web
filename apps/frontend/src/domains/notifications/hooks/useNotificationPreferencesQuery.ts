import { useQuery } from '@tanstack/react-query';

import {
  fetchNotificationPreferences,
  notificationPreferencesQueryKey,
} from '../api/notificationPreferences';

export const useNotificationPreferencesQuery = (userId: string | null) => {
  return useQuery({
    queryKey: notificationPreferencesQueryKey(userId),
    enabled: Boolean(userId),
    staleTime: 60_000,
    queryFn: async () => {
      if (!userId) {
        throw new Error('User ID is required to load notification preferences.');
      }

      return fetchNotificationPreferences(userId);
    },
  });
};
