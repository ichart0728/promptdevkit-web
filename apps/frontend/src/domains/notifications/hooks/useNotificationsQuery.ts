import { useInfiniteQuery } from '@tanstack/react-query';

import { supabase } from '@/lib/supabase';

import type { NotificationItem } from '../types';

export const NOTIFICATIONS_PAGE_SIZE = 20;

export const notificationsQueryKey = (userId: string | null) => ['notifications', userId] as const;

export const useNotificationsQuery = (userId: string | null) => {
  return useInfiniteQuery({
    queryKey: notificationsQueryKey(userId),
    initialPageParam: 0,
    enabled: Boolean(userId),
    staleTime: 60_000,
    queryFn: async ({ pageParam }) => {
      if (!userId) {
        return [] as NotificationItem[];
      }

      const page = typeof pageParam === 'number' ? pageParam : 0;
      const from = page * NOTIFICATIONS_PAGE_SIZE;
      const to = from + NOTIFICATIONS_PAGE_SIZE - 1;

      const { data, error } = await supabase
        .from('notifications')
        .select('id, type, payload, read_at, created_at')
        .eq('user_id', userId)
        .order('created_at', { ascending: false })
        .range(from, to);

      if (error) {
        throw error;
      }

      return (data ?? []) as NotificationItem[];
    },
    getNextPageParam: (lastPage, allPages) =>
      lastPage.length === NOTIFICATIONS_PAGE_SIZE ? allPages.length : undefined,
  });
};
