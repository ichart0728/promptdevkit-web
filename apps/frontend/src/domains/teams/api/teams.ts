import { queryOptions } from '@tanstack/react-query';

import { supabase } from '@/lib/supabase';

export type TeamMemberRole = 'admin' | 'editor' | 'viewer';

export type TeamMember = {
  id: string;
  role: TeamMemberRole;
  joinedAt: string;
  user: {
    id: string;
    email: string;
    name: string;
    avatarUrl: string | null;
  } | null;
};

export type Team = {
  id: string;
  name: string;
  createdAt: string;
  createdBy: string;
  planId: string | null;
  members: TeamMember[];
};

type TeamMemberUserRow = {
  id: string;
  email: string;
  name: string;
  avatar_url: string | null;
};

type TeamMemberRow = {
  id: string;
  role: TeamMemberRole;
  joined_at: string;
  user: TeamMemberUserRow | null;
};

type TeamOwnerPlanRow = { plan_id: string } | { plan_id: string }[] | null;

type TeamRow = {
  id: string;
  name: string;
  created_at: string;
  created_by: string;
  created_by_user: {
    user_plan: TeamOwnerPlanRow;
  } | null;
  team_members: TeamMemberRow[] | null;
};

const mapOwnerPlan = (plan: TeamOwnerPlanRow): string | null => {
  if (!plan) {
    return null;
  }

  if (Array.isArray(plan)) {
    return plan.length > 0 ? plan[0]?.plan_id ?? null : null;
  }

  return plan.plan_id ?? null;
};

const mapMember = (row: TeamMemberRow): TeamMember => ({
  id: row.id,
  role: row.role,
  joinedAt: row.joined_at,
  user: row.user
    ? {
        id: row.user.id,
        email: row.user.email,
        name: row.user.name,
        avatarUrl: row.user.avatar_url,
      }
    : null,
});

const mapRowToTeam = (row: TeamRow): Team => ({
  id: row.id,
  name: row.name,
  createdAt: row.created_at,
  createdBy: row.created_by,
  planId: mapOwnerPlan(row.created_by_user?.user_plan ?? null),
  members: (row.team_members ?? []).map(mapMember),
});

// Query key: ['teams', userId]
export const teamsQueryKey = (userId: string | null) => ['teams', userId ?? 'anonymous'] as const;

export const fetchTeams = async (): Promise<Team[]> => {
  const { data, error } = await supabase
    .from('teams')
    .select(
      `id,name,created_at,created_by,
       created_by_user:users!teams_created_by_fkey(
         user_plan:user_plans(plan_id)
       ),
       team_members(
         id,role,joined_at,
         user:users(
           id,email,name,avatar_url
         )
       )`
    )
    .order('created_at', { ascending: true });

  if (error) {
    throw error;
  }

  const rows = (data ?? []) as TeamRow[];

  return rows.map(mapRowToTeam);
};

export const teamsQueryOptions = (userId: string | null) =>
  queryOptions({
    queryKey: teamsQueryKey(userId),
    queryFn: async () => {
      if (!userId) {
        throw new Error('Cannot fetch teams without an authenticated user.');
      }

      return fetchTeams();
    },
    staleTime: 60 * 1000,
  });

type UpdateTeamMemberRoleParams = {
  teamId: string;
  memberId: string;
  role: TeamMemberRole;
};

export const updateTeamMemberRole = async ({
  teamId,
  memberId,
  role,
}: UpdateTeamMemberRoleParams): Promise<void> => {
  const { error } = await supabase
    .from('team_members')
    .update({ role } as never)
    .eq('team_id', teamId)
    .eq('id', memberId);

  if (error) {
    throw error;
  }
};

type RemoveTeamMemberParams = {
  teamId: string;
  memberId: string;
};

export const removeTeamMember = async ({
  teamId,
  memberId,
}: RemoveTeamMemberParams): Promise<void> => {
  const { error } = await supabase
    .from('team_members')
    .delete()
    .eq('team_id', teamId)
    .eq('id', memberId);

  if (error) {
    throw error;
  }
};
