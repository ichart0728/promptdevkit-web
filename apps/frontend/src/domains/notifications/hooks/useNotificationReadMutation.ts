import { useMutation, useQueryClient, type InfiniteData } from '@tanstack/react-query';

import { supabase } from '@/lib/supabase';

import type { NotificationItem } from '../types';
import { notificationsQueryKey } from './useNotificationsQuery';

type TogglePayload = {
  id: string;
  read: boolean;
};

type NotificationsInfiniteData = InfiniteData<NotificationItem[], number>;

type MutationError = {
  code?: string;
  message?: string;
};

export const useNotificationReadMutation = (userId: string | null) => {
  const queryClient = useQueryClient();
  const resolvedUserId = userId ?? null;
  const queryKey = notificationsQueryKey(resolvedUserId);

  const toggleMutation = useMutation({
    mutationFn: async ({ id, read }: TogglePayload) => {
      if (!resolvedUserId) {
        throw new Error('User ID is required to update notifications.');
      }

      const readAt = read ? new Date().toISOString() : null;

      const { error } = await supabase
        .from('notifications')
        .update({ read_at: readAt } as never)
        .eq('id', id)
        .eq('user_id', resolvedUserId);

      if (error) {
        throw error;
      }

      return { id, read_at: readAt };
    },
    onSuccess: ({ id, read_at }) => {
      queryClient.setQueryData(queryKey, (previous: NotificationsInfiniteData | undefined) => {
        if (!previous) {
          return previous;
        }

        return {
          ...previous,
          pages: previous.pages.map((page) =>
            page.map((notification) =>
              notification.id === id ? { ...notification, read_at } : notification,
            ),
          ),
        };
      });
    },
    onSettled: () => {
      if (resolvedUserId) {
        void queryClient.invalidateQueries({ queryKey });
      }
    },
  });

  const markAllMutation = useMutation({
    mutationFn: async () => {
      if (!resolvedUserId) {
        throw new Error('User ID is required to update notifications.');
      }

      const readAt = new Date().toISOString();

      const { error } = await supabase
        .from('notifications')
        .update({ read_at: readAt } as never)
        .eq('user_id', resolvedUserId)
        .is('read_at', null);

      if (error) {
        const mutationError = error as MutationError;

        if (mutationError.code === 'P0001') {
          throw new Error('You do not have permission to mark all notifications as read.');
        }

        throw error;
      }

      return readAt;
    },
    onSuccess: (readAt) => {
      queryClient.setQueryData(queryKey, (previous: NotificationsInfiniteData | undefined) => {
        if (!previous) {
          return previous;
        }

        return {
          ...previous,
          pages: previous.pages.map((page) =>
            page.map((notification) =>
              notification.read_at ? notification : { ...notification, read_at: readAt },
            ),
          ),
        };
      });
    },
    onSettled: () => {
      if (resolvedUserId) {
        void queryClient.invalidateQueries({ queryKey });
      }
    },
  });

  return {
    ...toggleMutation,
    mutateAll: markAllMutation.mutate,
    mutateAllAsync: markAllMutation.mutateAsync,
    isMutatingAll: markAllMutation.isPending,
    markAllError: markAllMutation.error,
  };
};
