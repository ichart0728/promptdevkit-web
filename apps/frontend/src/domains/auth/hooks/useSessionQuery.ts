import { useQuery } from '@tanstack/react-query';

import { sessionQueryOptions } from '../api/session';

export const useSessionQuery = () => useQuery(sessionQueryOptions);
