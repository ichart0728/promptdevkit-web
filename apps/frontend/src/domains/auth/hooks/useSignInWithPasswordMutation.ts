import { useMutation, useQueryClient, type UseMutationOptions } from '@tanstack/react-query';
import type { Session } from '@supabase/supabase-js';

import { signInWithPassword, type SignInWithPasswordParams } from '../api/sign-in';
import { sessionQueryOptions } from '../api/session';

export const useSignInWithPasswordMutation = (
  options?: Omit<UseMutationOptions<Session | null, Error, SignInWithPasswordParams>, 'mutationFn'>,
) => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: signInWithPassword,
    ...options,
    onSuccess: (session, variables, context, mutation) => {
      queryClient.setQueryData(sessionQueryOptions.queryKey, session);
      options?.onSuccess?.(session, variables, context, mutation);
    },
  });
};
