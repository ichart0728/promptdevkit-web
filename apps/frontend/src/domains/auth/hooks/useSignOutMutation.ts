import { useMutation, useQueryClient, type UseMutationOptions } from '@tanstack/react-query';

import { signOut } from '../api/sign-out';
import { sessionQueryOptions } from '../api/session';

export const useSignOutMutation = (
  options?: Omit<UseMutationOptions<void, Error, void>, 'mutationFn'>,
) => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: signOut,
    ...options,
    onSuccess: (data, variables, context, mutation) => {
      queryClient.setQueryData(sessionQueryOptions.queryKey, null);
      options?.onSuccess?.(data, variables, context, mutation);
    },
  });
};
