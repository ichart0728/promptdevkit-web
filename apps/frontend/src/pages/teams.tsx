import * as React from 'react';
import { useQuery } from '@tanstack/react-query';

import { useSessionQuery } from '@/domains/auth/hooks/useSessionQuery';
import { teamsQueryOptions } from '@/domains/teams/api/teams';
import { workspacesQueryOptions } from '@/domains/workspaces/api/workspaces';

const PLAN_LABELS: Record<string, string> = {
  free: 'Free',
  pro: 'Pro',
};

const getPlanLabel = (planId: string | null) => {
  if (!planId) {
    return 'Unknown plan';
  }

  return PLAN_LABELS[planId] ?? planId;
};

export const TeamsPage = () => {
  const sessionQuery = useSessionQuery();
  const userId = sessionQuery.data?.user?.id ?? null;

  const teamsQuery = useQuery({
    ...teamsQueryOptions(userId),
    enabled: !!userId,
  });

  const workspacesQuery = useQuery({
    ...workspacesQueryOptions(userId),
    enabled: !!userId,
  });

  const isSessionLoading = sessionQuery.status === 'pending';
  const isTeamsLoading = teamsQuery.status === 'pending';
  const isWorkspacesLoading = workspacesQuery.status === 'pending';
  const workspacesError = workspacesQuery.status === 'error';

  const teamWorkspaces = React.useMemo(() => {
    const entries = new Map<string, { id: string; name: string }[]>();

    if (workspacesQuery.status !== 'success') {
      return entries;
    }

    workspacesQuery.data
      .filter((workspace) => workspace.type === 'team' && workspace.teamId)
      .forEach((workspace) => {
        const existing = entries.get(workspace.teamId!);
        const next = [
          ...(existing ?? []),
          { id: workspace.id, name: workspace.name },
        ];

        entries.set(workspace.teamId!, next);
      });

    return entries;
  }, [workspacesQuery.status, workspacesQuery.data]);

  if (isSessionLoading) {
    return (
      <section className="space-y-4">
        <div>
          <h1 className="text-3xl font-bold">Teams</h1>
          <p className="text-muted-foreground">
            Invite collaborators and assign permissions across your organization.
          </p>
        </div>
        <div className="rounded-lg border border-dashed p-8 text-center text-sm text-muted-foreground">
          Loading teams…
        </div>
      </section>
    );
  }

  if (!userId) {
    return (
      <section className="space-y-4">
        <div>
          <h1 className="text-3xl font-bold">Teams</h1>
          <p className="text-muted-foreground">
            Invite collaborators and assign permissions across your organization.
          </p>
        </div>
        <div className="rounded-lg border border-dashed p-8 text-center text-sm text-muted-foreground">
          Sign in to view the teams you collaborate with.
        </div>
      </section>
    );
  }

  if (isTeamsLoading) {
    return (
      <section className="space-y-4">
        <div>
          <h1 className="text-3xl font-bold">Teams</h1>
          <p className="text-muted-foreground">
            Invite collaborators and assign permissions across your organization.
          </p>
        </div>
        <div className="rounded-lg border border-dashed p-8 text-center text-sm text-muted-foreground">
          Loading teams…
        </div>
      </section>
    );
  }

  if (teamsQuery.status === 'error') {
    const errorMessage =
      teamsQuery.error instanceof Error ? teamsQuery.error.message : 'Please try again.';

    return (
      <section className="space-y-4">
        <div>
          <h1 className="text-3xl font-bold">Teams</h1>
          <p className="text-muted-foreground">
            Invite collaborators and assign permissions across your organization.
          </p>
        </div>
        <div className="rounded-lg border border-destructive bg-destructive/10 p-8 text-center text-sm text-destructive">
          Failed to load teams. {errorMessage}
        </div>
      </section>
    );
  }

  const teams = teamsQuery.data ?? [];

  if (teams.length === 0) {
    return (
      <section className="space-y-4">
        <div>
          <h1 className="text-3xl font-bold">Teams</h1>
          <p className="text-muted-foreground">
            Invite collaborators and assign permissions across your organization.
          </p>
        </div>
        <div className="rounded-lg border border-dashed p-8 text-center text-sm text-muted-foreground">
          You’re not a member of any teams yet. Ask an administrator to invite you.
        </div>
      </section>
    );
  }

  return (
    <section className="space-y-6">
      <div className="space-y-2">
        <h1 className="text-3xl font-bold">Teams</h1>
        <p className="text-muted-foreground">
          Review membership and shared workspaces for the teams you belong to.
        </p>
      </div>
      <div className="space-y-6">
        {teams.map((team) => {
          const planLabel = getPlanLabel(team.planId);
          const members = team.members;
          const workspaces = teamWorkspaces.get(team.id) ?? [];

          return (
            <article key={team.id} className="rounded-lg border bg-card text-card-foreground shadow-sm">
              <header className="flex flex-col gap-4 border-b px-6 py-5 sm:flex-row sm:items-center sm:justify-between">
                <div>
                  <h2 className="text-xl font-semibold">{team.name}</h2>
                  <p className="text-sm text-muted-foreground">
                    {members.length} {members.length === 1 ? 'member' : 'members'}
                  </p>
                </div>
                <span className="inline-flex items-center rounded-full bg-primary/10 px-3 py-1 text-xs font-medium uppercase tracking-wide text-primary">
                  {planLabel}
                </span>
              </header>
              <div className="grid gap-6 px-6 py-5 md:grid-cols-2">
                <section aria-label={`${team.name} members`} className="space-y-3">
                  <h3 className="text-sm font-semibold uppercase text-muted-foreground">Members</h3>
                  <ul className="space-y-3">
                    {members.map((member) => (
                      <li key={member.id} className="flex items-start justify-between gap-3 rounded-md border p-3">
                        <div>
                          <p className="font-medium">{member.user?.name ?? 'Unknown member'}</p>
                          <p className="text-sm text-muted-foreground">
                            {member.user?.email ?? 'Contact information unavailable'}
                          </p>
                        </div>
                        <span className="inline-flex items-center rounded-full bg-muted px-2 py-0.5 text-xs font-medium capitalize text-muted-foreground">
                          {member.role}
                        </span>
                      </li>
                    ))}
                  </ul>
                </section>
                <section aria-label={`${team.name} shared workspaces`} className="space-y-3">
                  <h3 className="text-sm font-semibold uppercase text-muted-foreground">Shared workspaces</h3>
                  {isWorkspacesLoading ? (
                    <p className="text-sm text-muted-foreground">Loading shared workspaces…</p>
                  ) : workspacesError ? (
                    <p className="text-sm text-destructive">Failed to load shared workspaces. Please try again.</p>
                  ) : workspaces.length > 0 ? (
                    <ul className="space-y-3">
                      {workspaces.map((workspace) => (
                        <li key={workspace.id} className="rounded-md border p-3">
                          <p className="font-medium">{workspace.name}</p>
                          <p className="text-sm text-muted-foreground">Team workspace</p>
                        </li>
                      ))}
                    </ul>
                  ) : (
                    <p className="text-sm text-muted-foreground">No shared workspaces yet.</p>
                  )}
                </section>
              </div>
            </article>
          );
        })}
      </div>
    </section>
  );
};
