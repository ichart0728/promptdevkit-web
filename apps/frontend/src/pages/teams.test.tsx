import { cleanup, render, screen, waitFor } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { vi } from 'vitest';

import type { Team } from '@/domains/teams/api/teams';
import type { Workspace } from '@/domains/workspaces/api/workspaces';
import { TeamsPage } from './teams';
import { useSessionQuery } from '@/domains/auth/hooks/useSessionQuery';

const fetchPlanLimitsMock = vi.fn(async () => ({
  members_per_team: {
    key: 'members_per_team',
    value_int: 5,
    value_str: null,
    value_json: null,
  },
}));

const fetchTeamsMock = vi.fn(async () => [] as Team[]);
const fetchWorkspacesMock = vi.fn(async () => [] as Workspace[]);

vi.mock('@/domains/teams/api/teams', () => ({
  teamsQueryOptions: (userId: string | null) => ({
    queryKey: ['teams', userId ?? 'anonymous'] as const,
    queryFn: () => fetchTeamsMock(),
  }),
  teamsQueryKey: (userId: string | null) => ['teams', userId ?? 'anonymous'] as const,
  fetchTeams: () => fetchTeamsMock(),
  updateTeamMemberRole: vi.fn(),
  removeTeamMember: vi.fn(),
}));

vi.mock('@/domains/workspaces/api/workspaces', () => ({
  workspacesQueryOptions: (userId: string | null) => ({
    queryKey: ['workspaces', userId ?? 'anonymous'] as const,
    queryFn: () => fetchWorkspacesMock(),
  }),
  fetchWorkspaces: () => fetchWorkspacesMock(),
}));

vi.mock('@/domains/prompts/api/planLimits', () => ({
  planLimitsQueryKey: (planId: string) => ['plan-limits', planId] as const,
  fetchPlanLimits: (...args: Parameters<typeof fetchPlanLimitsMock>) => fetchPlanLimitsMock(...args),
}));

vi.mock('@/lib/supabase', () => ({
  supabase: {
    from: vi.fn(),
  },
}));

vi.mock('@/domains/auth/hooks/useSessionQuery', () => ({
  useSessionQuery: vi.fn(),
}));

const useSessionQueryMock = vi.mocked(useSessionQuery);

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: { retry: false },
    },
  });

const buildSessionQueryValue = (overrides: Partial<ReturnType<typeof useSessionQuery>> = {}) =>
  ({
    data: { user: { id: 'user-123', email: 'demo@example.com' } },
    status: 'success',
    isPending: false,
    ...overrides,
  } as unknown as ReturnType<typeof useSessionQuery>);

const renderTeamsPage = () => {
  const queryClient = createTestQueryClient();

  const renderResult = render(
    <QueryClientProvider client={queryClient}>
      <TeamsPage />
    </QueryClientProvider>,
  );

  return { ...renderResult, queryClient };
};

describe('TeamsPage', () => {
  beforeEach(() => {
    vi.clearAllMocks();
    useSessionQueryMock.mockReturnValue(buildSessionQueryValue());
    fetchWorkspacesMock.mockResolvedValue([]);
    fetchPlanLimitsMock.mockResolvedValue({
      members_per_team: {
        key: 'members_per_team',
        value_int: 5,
        value_str: null,
        value_json: null,
      },
    });
  });

  afterEach(() => {
    cleanup();
  });

  it('renders a loading state while teams are fetching', () => {
    fetchTeamsMock.mockReturnValue(new Promise<Team[]>(() => {}));

    renderTeamsPage();

    expect(screen.getByText('Loading teams…')).toBeInTheDocument();
  });

  it('renders the empty state when no teams exist', async () => {
    fetchTeamsMock.mockResolvedValue([]);

    renderTeamsPage();

    await waitFor(() => {
      expect(
        screen.getByText('You’re not a member of any teams yet. Ask an administrator to invite you.'),
      ).toBeInTheDocument();
    });
  });

  it('renders an error state when the query fails', async () => {
    const consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {});
    fetchTeamsMock.mockRejectedValueOnce(new Error('Network error'));

    renderTeamsPage();

    await screen.findByText('Failed to load teams. Network error');

    consoleErrorSpy.mockRestore();
  });

  it('renders team membership and workspaces', async () => {
    const team: Team = {
      id: 'team-1',
      name: 'Prompt Builders',
      createdAt: '2024-01-05T09:00:00Z',
      createdBy: 'user-123',
      planId: 'pro',
      members: [
        {
          id: 'member-1',
          role: 'admin',
          joinedAt: '2024-01-05T09:05:00Z',
          user: {
            id: 'user-123',
            name: 'Team Owner',
            email: 'team.owner@example.com',
            avatarUrl: null,
          },
        },
        {
          id: 'member-2',
          role: 'editor',
          joinedAt: '2024-01-05T09:06:00Z',
          user: {
            id: 'user-456',
            name: 'Team Editor',
            email: 'team.editor@example.com',
            avatarUrl: null,
          },
        },
      ],
    };

    fetchTeamsMock.mockResolvedValue([team]);
    fetchWorkspacesMock.mockResolvedValue([
      {
        id: 'workspace-1',
        name: 'Prompt Builders HQ',
        type: 'team',
        teamId: 'team-1',
        archivedAt: null,
      },
    ]);

    const { asFragment } = renderTeamsPage();

    await waitFor(() => {
      expect(screen.getByText('Prompt Builders')).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByText('This team is using 2 of 5 member seats.')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
