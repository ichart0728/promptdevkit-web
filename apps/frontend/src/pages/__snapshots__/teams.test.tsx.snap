// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TeamsPage > renders team membership and workspaces 1`] = `
<DocumentFragment>
  <section
    class="space-y-6"
  >
    <div
      class="space-y-2"
    >
      <h1
        class="text-3xl font-bold"
      >
        Teams
      </h1>
      <p
        class="text-muted-foreground"
      >
        Review membership and shared workspaces for the teams you belong to.
      </p>
    </div>
    <div
      class="space-y-6"
    >
      <article
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
      >
        <header
          class="flex flex-col gap-4 border-b px-6 py-5 sm:flex-row sm:items-center sm:justify-between"
        >
          <div>
            <h2
              class="text-xl font-semibold"
            >
              Prompt Builders
            </h2>
            <p
              class="text-sm text-muted-foreground"
            >
              2 members
            </p>
          </div>
          <span
            class="inline-flex items-center rounded-full bg-primary/10 px-3 py-1 text-xs font-medium uppercase tracking-wide text-primary"
          >
            Pro
          </span>
        </header>
        <div
          class="grid gap-6 px-6 py-5 md:grid-cols-2"
        >
          <section
            aria-label="Prompt Builders members"
            class="space-y-3"
          >
            <h3
              class="text-sm font-semibold uppercase text-muted-foreground"
            >
              Members
            </h3>
            <ul
              class="space-y-3"
            >
              <li
                class="flex flex-col gap-3 rounded-md border p-3 sm:flex-row sm:items-center sm:justify-between"
              >
                <div>
                  <p
                    class="font-medium"
                  >
                    Team Owner
                  </p>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    team.owner@example.com
                  </p>
                </div>
                <div
                  class="flex flex-col items-stretch gap-2 sm:flex-row sm:items-center sm:gap-3"
                >
                  <div
                    class="flex flex-col gap-1 text-left sm:text-right"
                  >
                    <label
                      class="sr-only"
                      for=":r0:"
                    >
                      Role
                    </label>
                    <select
                      aria-label="Role"
                      class="h-9 rounded-md border border-input bg-background px-2 text-sm shadow-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
                      id=":r0:"
                    >
                      <option
                        disabled=""
                        value="viewer"
                      >
                        Viewer
                      </option>
                      <option
                        disabled=""
                        value="editor"
                      >
                        Editor
                      </option>
                      <option
                        value="admin"
                      >
                        Admin
                      </option>
                    </select>
                  </div>
                  <button
                    class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                  >
                    Leave team
                  </button>
                </div>
              </li>
              <li
                class="flex flex-col gap-3 rounded-md border p-3 sm:flex-row sm:items-center sm:justify-between"
              >
                <div>
                  <p
                    class="font-medium"
                  >
                    Team Editor
                  </p>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    team.editor@example.com
                  </p>
                </div>
                <div
                  class="flex flex-col items-stretch gap-2 sm:flex-row sm:items-center sm:gap-3"
                >
                  <div
                    class="flex flex-col gap-1 text-left sm:text-right"
                  >
                    <label
                      class="sr-only"
                      for=":r1:"
                    >
                      Role
                    </label>
                    <select
                      aria-label="Role"
                      class="h-9 rounded-md border border-input bg-background px-2 text-sm shadow-sm focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
                      id=":r1:"
                    >
                      <option
                        value="viewer"
                      >
                        Viewer
                      </option>
                      <option
                        value="editor"
                      >
                        Editor
                      </option>
                      <option
                        value="admin"
                      >
                        Admin
                      </option>
                    </select>
                  </div>
                  <button
                    class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-9 rounded-md px-3"
                  >
                    Remove member
                  </button>
                </div>
              </li>
            </ul>
          </section>
          <section
            aria-label="Prompt Builders shared workspaces"
            class="space-y-3"
          >
            <h3
              class="text-sm font-semibold uppercase text-muted-foreground"
            >
              Shared workspaces
            </h3>
            <ul
              class="space-y-3"
            >
              <li
                class="rounded-md border p-3"
              >
                <p
                  class="font-medium"
                >
                  Prompt Builders HQ
                </p>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Team workspace
                </p>
              </li>
            </ul>
          </section>
        </div>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
